x/evm/keeper/msg_ethereum_tx_test.go

```diff
@@ -34,12 +49,37 @@ func (s *Suite) TestMsgEthereumTx_CreateContract() {
 					GasLimit:      gasLimit,
 				}
 				ethTxMsg, err := evmtest.CreateContractTxMsg(args)
-				s.NoError(err)
+				s.Require().NoError(err)
 				s.Require().NoError(ethTxMsg.ValidateBasic())
 				s.Equal(ethTxMsg.GetGas(), gasLimit.Uint64())
 
 				resp, err := deps.Chain.EvmKeeper.EthereumTx(deps.GoCtx(), ethTxMsg)
-				s.Require().NoError(err, "resp: %s\nblock header: %s", resp, deps.Ctx.BlockHeader().ProposerAddress)
+				s.Require().NoError(
+					err,
+					"resp: %s\nblock header: %s",
+					resp,
+					deps.Ctx.BlockHeader().ProposerAddress,
+				)
+				s.Require().Empty(resp.VmError)
+
+				// Event "EventContractDeployed" must present
+				var sdkEvents = deps.Ctx.EventManager().Events()
+				contractDeployedEventType := "eth.evm.v1.EventContractDeployed"
+				err = testutil.AssertEventPresent(sdkEvents, contractDeployedEventType)
+				s.Require().NoError(err)
+
+				var contractDeployedEvent sdk.Event
+				for _, abciEvent := range sdkEvents {
+					if abciEvent.Type == contractDeployedEventType {
+						contractDeployedEvent = abciEvent
+					}
+				}
+				for _, err = range []error{
+					testutil.EventHasAttributeValue(contractDeployedEvent, "sender", ethAcc.EthAddr.String()),
+					testutil.EventHasAttributeValue(contractDeployedEvent, "contract_addr", resp.Logs[0].Address),
+				} {
+					s.Require().NoError(err)
```

Comment:
You can probably simplify this with 
`func RequireContainsTypedEvent(t require.TestingT, ctx sdk.Context, event proto.Message) `